iib/workers/tasks/build.py

```diff
@@ -876,6 +918,248 @@ def get_image_label(pull_spec, label):
+    :param str ocp_version: ocp version by which to filter the bundles.
+    :return: bundles which don't satisfy the ocp version.
+    :rtype: list
+    """
+    ocp_version = [int(num) for num in ocp_version.replace('v', '').split('.')]
+    deprecate_bundles = []
+    for bundle, metadata in bundles_with_metadata:
+        bundle_ocp_version = metadata['Labels']['com.redhat.openshift.versions']
+        # only one version
+        if bundle_ocp_version[0] == '=':
+            accepted_version = [
+                int(num) for num in bundle_ocp_version.replace('=', '').replace('v', '').split('.')
+            ]
+            if ocp_version != accepted_version:
+                deprecate_bundles.append(bundle)
+        # version range
+        elif '-' in bundle_ocp_version:
+            min_version, max_version = [
+                ver.split('.') for ver in bundle_ocp_version.replace('v', '').split('-')
+            ]
+            min_version = [int(num) for num in min_version]
+            max_version = [int(num) for num in max_version]
+            if ocp_version < min_version or ocp_version > max_version:
+                deprecate_bundles.append(bundle)
+        # multiple csv versions
+        elif ',' in bundle_ocp_version:
+            str_versions = [
+                version.split('.') for version in bundle_ocp_version.replace('v', '').split(',')
+            ]
+            versions = []
+            for version in str_versions:
+                versions.append([int(num) for num in version])
+            min_version = sorted(versions)[-1]
+            if ocp_version < min_version:
+                deprecate_bundles.append(bundle)
+        # minimum version
+        else:
+            min_version = [int(num) for num in bundle_ocp_version.replace('v', '').split('.')]
+            if ocp_version < min_version:
+                deprecate_bundles.append(bundle)
+
+    return deprecate_bundles
+
+
+def _check_operator_heads_for_deprecation(bundles, ocp_version):
+    """
+    Calculate the head of each operator and determine whether they should be deprecated.
+
+    :param list bundles: list of bundles to extract the data from.
+    :param str ocp_version: ocp version to classify deprecation.
+    :return: list of bundles to be deprecated.
+    :rtype: list
+    """
+    operators_bundles_asoc = {}
+    for bundle in bundles:
+        if bundle['channelName'] not in operators_bundles_asoc:
+            operators_bundles_asoc[bundle['channelName']] = []
+        operators_bundles_asoc[bundle['channelName']].append(bundle)
+
+    channel_heads = []
+    for bundles in operators_bundles_asoc.values():
+        # comparing versions as strings seems the safest
+        channel_heads.append(sorted(bundles, key=lambda x: x['version'], reverse=True)[0])
+
+    bundles_with_metadata = []
+    for head_bundle in channel_heads:
+        skopeo_out = skopeo_inspect('docker://{0}'.format(head_bundle['bundlePath']))
```

Comment:
`skopeo_inspect` is an expensive operation. Since you're simply using a single label, could you use the get_image_label method to get it? (It uses the --config option which is extremely fast). Check out https://github.com/release-engineering/iib/commit/f58767585bfedbbb4d771c9f238d23d27a1aeaf6 for reference. If you're using it in any other place, could you please replace those too?

Reply:
Fixed.